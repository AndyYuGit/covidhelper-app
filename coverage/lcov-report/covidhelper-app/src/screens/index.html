
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for covidhelper-app/src/screens</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> covidhelper-app/src/screens</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">18.9% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>55/291</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">2.85% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>4/140</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">14.1% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>11/78</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">19.36% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>55/284</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line low'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="ChatbotScreen.js"><a href="ChatbotScreen.js.html">ChatbotScreen.js</a></td>
	<td data-value="34.61" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 34%"></div><div class="cover-empty" style="width: 66%"></div></div>
	</td>
	<td data-value="34.61" class="pct low">34.61%</td>
	<td data-value="26" class="abs low">9/26</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="12" class="abs low">3/12</td>
	<td data-value="37.5" class="pct low">37.5%</td>
	<td data-value="24" class="abs low">9/24</td>
	</tr>

<tr>
	<td class="file low" data-value="ContactScreen.js"><a href="ContactScreen.js.html">ContactScreen.js</a></td>
	<td data-value="24.13" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 24%"></div><div class="cover-empty" style="width: 76%"></div></div>
	</td>
	<td data-value="24.13" class="pct low">24.13%</td>
	<td data-value="29" class="abs low">7/29</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	<td data-value="12.5" class="pct low">12.5%</td>
	<td data-value="8" class="abs low">1/8</td>
	<td data-value="28" class="pct low">28%</td>
	<td data-value="25" class="abs low">7/25</td>
	</tr>

<tr>
	<td class="file low" data-value="DistrictsScreen.js"><a href="DistrictsScreen.js.html">DistrictsScreen.js</a></td>
	<td data-value="18.18" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 18%"></div><div class="cover-empty" style="width: 82%"></div></div>
	</td>
	<td data-value="18.18" class="pct low">18.18%</td>
	<td data-value="22" class="abs low">4/22</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="5.26" class="pct low">5.26%</td>
	<td data-value="19" class="abs low">1/19</td>
	<td data-value="18.18" class="pct low">18.18%</td>
	<td data-value="22" class="abs low">4/22</td>
	</tr>

<tr>
	<td class="file medium" data-value="HomeScreen.js"><a href="HomeScreen.js.html">HomeScreen.js</a></td>
	<td data-value="55.55" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 55%"></div><div class="cover-empty" style="width: 45%"></div></div>
	</td>
	<td data-value="55.55" class="pct medium">55.55%</td>
	<td data-value="9" class="abs medium">5/9</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="12" class="abs low">3/12</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="3" class="abs low">1/3</td>
	<td data-value="55.55" class="pct medium">55.55%</td>
	<td data-value="9" class="abs medium">5/9</td>
	</tr>

<tr>
	<td class="file low" data-value="LocationsScreen.js"><a href="LocationsScreen.js.html">LocationsScreen.js</a></td>
	<td data-value="23.52" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 23%"></div><div class="cover-empty" style="width: 77%"></div></div>
	</td>
	<td data-value="23.52" class="pct low">23.52%</td>
	<td data-value="34" class="abs low">8/34</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	<td data-value="10" class="pct low">10%</td>
	<td data-value="10" class="abs low">1/10</td>
	<td data-value="23.52" class="pct low">23.52%</td>
	<td data-value="34" class="abs low">8/34</td>
	</tr>

<tr>
	<td class="file low" data-value="ProfileScreen.js"><a href="ProfileScreen.js.html">ProfileScreen.js</a></td>
	<td data-value="22.85" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 22%"></div><div class="cover-empty" style="width: 78%"></div></div>
	</td>
	<td data-value="22.85" class="pct low">22.85%</td>
	<td data-value="35" class="abs low">8/35</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="14" class="abs low">0/14</td>
	<td data-value="11.11" class="pct low">11.11%</td>
	<td data-value="9" class="abs low">1/9</td>
	<td data-value="22.85" class="pct low">22.85%</td>
	<td data-value="35" class="abs low">8/35</td>
	</tr>

<tr>
	<td class="file low" data-value="ReportScreen.js"><a href="ReportScreen.js.html">ReportScreen.js</a></td>
	<td data-value="10.29" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 10%"></div><div class="cover-empty" style="width: 90%"></div></div>
	</td>
	<td data-value="10.29" class="pct low">10.29%</td>
	<td data-value="136" class="abs low">14/136</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="100" class="abs low">0/100</td>
	<td data-value="17.64" class="pct low">17.64%</td>
	<td data-value="17" class="abs low">3/17</td>
	<td data-value="10.37" class="pct low">10.37%</td>
	<td data-value="135" class="abs low">14/135</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2023-01-29T14:49:26.276Z
            </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    