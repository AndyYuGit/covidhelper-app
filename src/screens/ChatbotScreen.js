import React, { useState, useCallback, useEffect } from 'react'
import { StyleSheet, Text, View ,Switch, TextInput,ScrollView,TouchableOpacity,Image,FlatList} from 'react-native';
import { GiftedChat,InputToolbar,LeftAction, ChatInput, SendButton,Bubble,Composer,Send } from 'react-native-gifted-chat'
import { axiosGet } from '../services/AxiosGet';
import{URLWithDeviceInfo} from '../common/URLWithDeviceInfo'
import {Logging} from '../common/Logging'


export function ChatbotScreen(props){
    const [messages, setMessages] = useState([]);
    const [information, setInformation] = useState([]);
    const [messageCount, setMessageCount] = useState(2);

    useEffect(() => {
        setMessages([
          {
            _id: 1,
            text: 'Hello, what can I help you?Following the adjustment of local anti-epidemic measures, starting from 30 December 2022, the Department of Health will not provide the list of buildings visited by cases tested positive for SARS-CoV-2 virus in the past 14 days. The update of latest visit building has temporarily terminated until further adjustment of local anti-epidemic measures. The version of latest visit building has preserved on 28 December 2022',
            createdAt: new Date(),
            user: {
              _id: 2,
              name: 'Bot',
              avatar: require('../../assets/bot-icon.png'),
            },
          },
        ])
      }, [])

      useEffect(() => {
        if(information.length!=0){
          setMessageCount(messageCount+1)
          Logging('Get information:'+information.toString(),'INFO')
          setMessages(previousMessages => GiftedChat.append(previousMessages, [
            {
              _id: messageCount,
              text: information.toString(),
              user: {
                 _id: 2,
                name: 'Bot',
                avatar: require('../../assets/bot-icon.png'),
              },
            }
          ]))
        }
      }, [information])

      async function getInformation(messageText){
        try {
          setInformation(await axiosGet(URLWithDeviceInfo('http://covidhelper1.ddns.net:8080/keywords/Information?message='+messageText)))
        } catch (error) {
          await Logging('Get information request by message:'+messageText+' error:','ERROR')
        }
      }


      async function onSend(messages){
        setMessages(previousMessages => GiftedChat.append(previousMessages, [
          {
            _id: messageCount,
            text: messages[0].text,
            user: {
               _id: 1,
            },
          }
        ]))
        setMessageCount(messageCount+1)
        await Logging('Get information request by message:'+messages[0].text,'INFO')
        await getInformation(messages[0].text)
      }

      function renderBubble(props) {
        return (
          <Bubble
            {...props}
            wrapperStyle={{
              left: {
                backgroundColor: '#cba3ce',
              },
              right:{
                backgroundColor: '#cba3ce',
              }
            }}
            textStyle={{
              left: {
                color:'white'
              },
              right: {
                color: 'white',
              }
            }}
            timeTextStyle={{ left: { color: 'white' },right: { color:'white'} }}
          />
        );
      }

      function renderSend(props){
        return(
          <Send
            {...props}
            textStyle={{color:'#cba3ce'}}
          />
        )
      }
     
      function renderInputToolbar(props) {
        return (
          <InputToolbar
            {...props}
            containerStyle={{
              backgroundColor: '#21254a',
              color:'white'
            }}
            renderComposer={props=>
              <Composer
                {...props} 
                textInputProps={{...props.textInputProps,selectionColor:'#cba3ce'}}
                textInputStyle={{ color: 'white'}} 
                placeholderTextColor = 'white' 
              /> 
            } 
          />
        );
      };
    
      return (
        <View style={styles.container}>
          <GiftedChat 
            messages={messages}
            onSend={messages => onSend(messages)}
            user={{
              _id: 1,
            }}
            renderBubble={renderBubble}
            renderInputToolbar={renderInputToolbar}
            renderSend={renderSend}
          />
        </View>
      )
}

const styles = StyleSheet.create({
  container: {
    flexGrow: 1,
    backgroundColor: '#EEF0F4',
    marginTop:-33,
  }
});