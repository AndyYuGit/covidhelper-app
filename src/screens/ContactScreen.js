import { StyleSheet, Text, View,Switch,TouchableOpacity, TextInput  } from 'react-native';
import React, { useState, useEffect} from "react";
import AsyncStorage from "@react-native-async-storage/async-storage";
import {Logging} from '../common/Logging'
import * as MediaLibrary from 'expo-media-library';
import * as FileSystem from 'expo-file-system';
import { axiosGet } from '../services/AxiosGet';
import{URLWithDeviceInfo} from '../common/URLWithDeviceInfo'
import * as SMS from 'expo-sms';
import * as Application from 'expo-application';
import Constants from "expo-constants"
import * as Device from 'expo-device';

export function ContactScreen (props){
    const [text, onChangeText] = useState("");
    const [warning, setWarning] = useState("");
    const [adminInfo,setAdminInfo]=useState("");
    const version = Constants.manifest.version

    useEffect(() => {
       const getAdminInfo = async () => {
          const result = await axiosGet(URLWithDeviceInfo('http://covidhelper1.ddns.net:8080/admins/AdminEmailPhoneNum?id=1'))
          setAdminInfo(result);
        };
        getAdminInfo()
    },[]);
  
    return(
        <View style={{flex: 1}}>
            <View style={styles.rows}>
            <Text style={styles.text}>Write something...</Text>
            <TextInput style ={styles.textBox} multiline placeholder="Tell me your advice or bug report..." selectionColor="#cba3ce" onChangeText={onChangeText}
            value={text}/>
            <TouchableOpacity key={'Send'} style={styles.button} onPress={async()=>{
                const mediaLibrarypermission=await MediaLibrary.requestPermissionsAsync()
                let logs=await AsyncStorage.getItem('logs')
                logs=logs!=null?JSON.parse(logs):[];
                const isAvailable = await SMS.isAvailableAsync();
                if (isAvailable&&mediaLibrarypermission) {
                    let fileUri = FileSystem.documentDirectory + "logs.txt";
                    await FileSystem.writeAsStringAsync(fileUri, 'deviceID='+Application.androidId+' model='+Device.modelName+' osVersion='+Device.osVersion+' appVersion='+version+'\n'+logs.toString(), { encoding: FileSystem.EncodingType.UTF8 });
                    const asset = await MediaLibrary.createAssetAsync(fileUri).then((response) => response).catch((error) => error)
                    await MediaLibrary.createAssetAsync(asset).then((response) => response).catch((error) => error)
                    const cURI=await FileSystem.getContentUriAsync(fileUri);
                    const { result } = await SMS.sendSMSAsync(
                        [adminInfo[1]],
                        text,
                        {
                            attachments: {
                              uri: cURI,
                              mimeType: 'text/plain',
                              filename: 'logs.txt',
                            },
                        }
                    )
                    await Logging('Send button is clicked with:'+text,'INFO')
                } else {
                    await Logging('Your SMS service is not available currently. Please try again or contact '+adminInfo[0]+' for help.','WARN')
                    setWarning('Your SMS service is not available currently. Please try againor contact '+adminInfo[0]+' for help.')
                }
            }}> 
             <Text style={styles.buttonLabel}>Send</Text>
            </TouchableOpacity>
            <Text style={styles.text}>{warning}</Text>
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    rows: {
        flex: 1,flexDirection:'column',
    },
    itemLeft:{
        flex: 1,paddingTop:10
    },
    itemRight:{
        flex: 1,
    },
    text:{
        color:'#282F6F',fontSize: 16,
    },
    row:{
        padding: 10,
    },
    textInItem:{
        color:'#21254a'
    },
    textBox:{
        maxHeight: 500, 
        borderColor: "#21254a", 
        borderWidth: 1, 
        padding: 10, 
        width: "95%",
    },
    button:{
        fontSize:14,
        backgroundColor:'#21254a',
        borderRadius:10,
        borderColor:'#cba3ce',
        borderWidth: 2.4,
        width: "15%",
    },
    buttonLabel:{
        color:'white',fontSize: 22,
    },
  });