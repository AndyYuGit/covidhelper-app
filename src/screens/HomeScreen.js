import { StyleSheet, Text, View ,ScrollView} from 'react-native';
import React, { useEffect, useState } from "react";
import { NavigationContainer, useIsFocused } from '@react-navigation/native';
import { LatestCases } from '../common/LatestCases';


export function HomeScreen (props){
    const [result,setResult]=useState(0);
    const isFocused = useIsFocused();

    useEffect(() => {
        const getResult = async () => {
          const result = await LatestCases();
          setResult(result);
        };
        getResult()
    },[]);

    

    return (
     <View style={styles.container}>
          <View style={[styles.circle]}>
            <Text style={styles.dataText}>{typeof result[0]==='undefined'?'Loading...':(isFocused ?result[0]:'')}</Text>
            <Text style={styles.labelText}>Confirmed Cases</Text>
            <Text>{"\n"}</Text>
            <Text style={styles.dataText}>{typeof result[1]==='undefined'?'Loading...':(isFocused ?result[1]:'')}</Text>
            <Text style={styles.labelText}>Death Cases</Text>
          </View>          
          <View style={styles.footer}>
            <Text style={styles.labelText}>Last updated at: {typeof result[2]==='undefined'?'':(isFocused?result[2].getFullYear()+'-'+(result[2].getMonth()+1)+'-'+result[2].getDate():'')}</Text>
          </View>
    </View>
      
    );
  };

  const styles = StyleSheet.create({
    container: {
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      backgroundColor: '#EEF0F4'
    },
    circle:{
      position:'absolute',
      height:300,
      width:300,
      backgroundColor:'#EEF0F4',
      borderRadius:150,
      elevation:7,
      justifyContent: 'center',
      alignItems: 'center',
    },
    footer: {
      position:'absolute',
      bottom:100
    },
    dataText:{
      color:'#cba3ce',fontSize:22
    },
    labelText:{
      color:'#21254a',fontSize:14
    }

    
  });