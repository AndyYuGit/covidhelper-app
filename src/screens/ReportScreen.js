import { StyleSheet, Text, View ,Switch, TextInput,ScrollView,TouchableOpacity,Image,FlatList} from 'react-native';
import React, { useEffect, useState } from "react";
import { axiosGet } from '../services/AxiosGet';
import { NavigationContainer, useIsFocused } from '@react-navigation/native';
import AsyncStorage from "@react-native-async-storage/async-storage";
import{URLWithDeviceInfo} from '../common/URLWithDeviceInfo'
import {Logging} from '../common/Logging'

function isDifferenceSmallerThan(num1,num2,target){
    if(num1>=num2){
        return num1-num2<=target?true:false;
    }
    else{
        return num2-num1<=target?true:false;
    }
}

export  function ReportScreen(props){
    const [profiles,setProfiles]=useState('');
    const [storedLocations,setStoredLocations]=useState(null);
    const [latestVisitBuildings,setLatestVisitBuildings]=useState(null);
    const [currentReport,setCurrentReport]=useState(0);
    const [currentReportName,setCurrentReportName]=useState('Choose a report');
    const [currentReportDate,setCurrentReportDate]=useState('');
    const [reportResult,setReportResult]=useState(null);
    const isFocused = useIsFocused();

    function fillInReport(){
        let tempReportResult=[]
        if(latestVisitBuildings==null){
            return;
        }
        latestVisitBuildings.forEach(building=>{
            let isLatitudeNear,isLongitudeNear;
            let isEqualProfile=false;
            switch(currentReport){
                case 1:
                    isLatitudeNear=isDifferenceSmallerThan(profiles[2],building.latitude,0.01)
                    isLongitudeNear=isDifferenceSmallerThan(profiles[3],building.longitude,0.01)
                    if(building.district===profiles[0]&&building.name===profiles[1]){
                        isEqualProfile=true;
                    }
                    break;
                case 2:
                    isLatitudeNear=isDifferenceSmallerThan(profiles[6],building.latitude,0.01)
                    isLongitudeNear=isDifferenceSmallerThan(profiles[7],building.longitude,0.01)
                    if(building.district===profiles[4]&&building.name===profiles[5]){
                        isEqualProfile=true;
                    }
                    break;
                case 3:
                    isLatitudeNear=isDifferenceSmallerThan(profiles[10],building.latitude,0.01)
                    isLongitudeNear=isDifferenceSmallerThan(profiles[11],building.longitude,0.01)
                    if(building.district===profiles[8]&&building.name===profiles[9]){
                        isEqualProfile=true;
                    }
                    break;
                case 4:
                    for(let i=0;i<storedLocations.length;i++){
                        isLatitudeNear=isDifferenceSmallerThan(storedLocations[i].latitude,building.latitude,0.01)
                        isLongitudeNear=isDifferenceSmallerThan(storedLocations[i].longitude,building.longitude,0.01)
                        if(isLatitudeNear&&isLongitudeNear){
                            break;
                        }
                    }
                    break;
            }
            if(isLatitudeNear&&isLongitudeNear){
                if(isEqualProfile){
                    let profileBuilding=[building]
                    tempReportResult=profileBuilding.concat(tempReportResult)
                }
                else{
                    tempReportResult.push(building)
                }
                setCurrentReportDate(building.latedate)
                
            }
        })
        setReportResult(tempReportResult)
    }

    function clickUp(){
        if(currentReport==1||currentReport==0){
            setCurrentReport(4)
            setCurrentReportName('Stored Locations')
        }
        else if(currentReport==2){
            setCurrentReport(1)
            setCurrentReportName(profiles[0]+' '+profiles[1])
        }
        else if(currentReport==3){
            setCurrentReport(2)
            setCurrentReportName(profiles[4]+' '+profiles[5])
        }
        else if(currentReport==4){
            setCurrentReport(3)
            setCurrentReportName(profiles[8]+' '+profiles[9])
        }
    }

    function clickDown(){
        if(currentReport==1){
            setCurrentReport(2)
            setCurrentReportName(profiles[4]+' '+profiles[5])
        }
        else if(currentReport==2){
            setCurrentReport(3)
            setCurrentReportName(profiles[8]+' '+profiles[9])
        }
        else if(currentReport==3){
            setCurrentReport(4)
            setCurrentReportName('Stored Locations')
        }
        else if(currentReport==4||currentReport==0){
            setCurrentReport(1)
            setCurrentReportName(profiles[0]+' '+profiles[1])
        }
    }

    function BuildingsSeperator(){
        return (
          <View style={styles.buildingsSeperator}/>
        );
    }
    function renderbuildings({item,index}){
        return (
            <Text style={styles.textInItem}>{item.district} {item.name}</Text>
        )
      }


    useEffect(()=>{
        const getProfiles =async () => {
            try {
                let districtOfProfile1=await AsyncStorage.getItem('districtOfProfile1')
                let districtOfProfile2=await AsyncStorage.getItem('districtOfProfile2')
                let districtOfProfile3=await AsyncStorage.getItem('districtOfProfile3')
                let buildingOfProfile1=await AsyncStorage.getItem('buildingOfProfile1')
                let buildingOfProfile2=await AsyncStorage.getItem('buildingOfProfile2')
                let buildingOfProfile3=await AsyncStorage.getItem('buildingOfProfile3')
                let latitudeOfProfile1=parseFloat(await AsyncStorage.getItem('latitudeOfProfile1'))
                let latitudeOfProfile2=parseFloat(await AsyncStorage.getItem('latitudeOfProfile2'))
                let latitudeOfProfile3=parseFloat(await AsyncStorage.getItem('latitudeOfProfile3'))
                let longitudeOfProfile1=parseFloat(await AsyncStorage.getItem('longitudeOfProfile1'))
                let longitudeOfProfile2=parseFloat(await AsyncStorage.getItem('longitudeOfProfile2'))
                let longitudeOfProfile3=parseFloat(await AsyncStorage.getItem('longitudeOfProfile3'))
                districtOfProfile1=districtOfProfile1 !=null?districtOfProfile1:'None'
                districtOfProfile2=districtOfProfile2 !=null?districtOfProfile2:'None'
                districtOfProfile3=districtOfProfile3 !=null?districtOfProfile3:'None'
                buildingOfProfile1=buildingOfProfile1 !=null?buildingOfProfile1:''
                buildingOfProfile2=buildingOfProfile2 !=null?buildingOfProfile2:''
                buildingOfProfile3=buildingOfProfile3 !=null?buildingOfProfile3:''
                latitudeOfProfile1=!isNaN(latitudeOfProfile1)?latitudeOfProfile1:''
                latitudeOfProfile2=!isNaN(latitudeOfProfile2)?latitudeOfProfile2:''
                latitudeOfProfile3=!isNaN(latitudeOfProfile3)?latitudeOfProfile3:''
                longitudeOfProfile1=!isNaN(longitudeOfProfile1)?longitudeOfProfile1:''
                longitudeOfProfile2=!isNaN(longitudeOfProfile2)?longitudeOfProfile2:''
                longitudeOfProfile3=!isNaN(longitudeOfProfile3)?longitudeOfProfile3:''
                setProfiles([districtOfProfile1,buildingOfProfile1,latitudeOfProfile1,longitudeOfProfile1,
                  districtOfProfile2,buildingOfProfile2,latitudeOfProfile2,longitudeOfProfile2,
                  districtOfProfile3,buildingOfProfile3,latitudeOfProfile3,longitudeOfProfile3])
            } catch (error) {
              console.log(error)
            }        
          };

          const getStoredLocations=async()=>{
            try {
                await setStoredLocations(JSON.parse(await AsyncStorage.getItem('storedLocations')))
            } catch (error) {
                console.log(error)
            }
          }
          const getLatestVisitBuildings=async()=>{
            try {
                setLatestVisitBuildings(await axiosGet(URLWithDeviceInfo('http://covidhelper1.ddns.net:8080/latestvisitbuilding/')))
            } catch (error) {
                console.log(error)
            }
          }

          const setInitialReport=async ()=>{
            if(profiles[0]!=null&&profiles[1]!=null&&!isNaN(profiles[2])&&isNaN(profiles[3])){
                await setCurrentReport(1)
                await setCurrentReportName(profiles[0]+' '+profiles[1])
            }
            else if(profiles[4]!=null&&profiles[5]!=null&&!isNaN(profiles[6])&&!isNaN(profiles[7])){
                await setCurrentReport(2)
                await setCurrentReportName(profiles[4]+' '+profiles[5])
            }
            else if(profiles[8]!=null&&profiles[9]!=null&&!isNaN(profiles[10])&&!isNaN(profiles[11])){
                await setCurrentReport(3)
                await setCurrentReportName(profiles[9]+' '+profiles[10])
            }
            else if(storedLocations!=null){
                await setCurrentReport(4)
                await setCurrentReportName('Stored Locations')
            }
            else{
                
            }
          }

          const runPreparation=async()=>{
            await getProfiles();
            await getStoredLocations()
            await getLatestVisitBuildings()
            await setInitialReport();
          }

          if(isFocused){
            runPreparation()
          }
           
    },[isFocused]);

    useEffect(() => {
        fillInReport()
        Logging('Switch to '+currentReportName+' '+currentReportDate,'INFO')  
    }, [currentReport])

    return(
        <View style={styles.container}>
            <View style={styles.profileBar}>
                <TouchableOpacity
                style={styles.button}
                onPress={clickUp}
                >
                    <Image source={require('../../assets/Button-Up-icon.png')} />
                </TouchableOpacity>
                <Text>{currentReportName}</Text>
                <TouchableOpacity
                style={styles.button}
                onPress={clickDown}
                >
                    <Image source={require('../../assets/Button-Down-icon.png')} />
                </TouchableOpacity>
            </View>
            <BuildingsSeperator/>
            <FlatList
            data={reportResult}
            keyExtractor={(item,index)=> index.toString()}
            ItemSeparatorComponent={BuildingsSeperator}
            renderItem={renderbuildings}
            contentContainerStyle={{flexGrow: 1}}
            ListHeaderComponent={()=><Text>Here are buildings about your selected locations with reported case at {currentReportDate}:</Text>}
            />
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
      flexGrow: 1,
      backgroundColor: '#EEF0F4',
      marginBottom:0.1
    },
    profileBar:{
        flex:1,
        flexDirection:'row',
    },
    buildingsSeperator:{
      height: 0.5,
      width: '100%',
      backgroundColor: '#cba3ce',
    },
    profileBar:{
      padding: 10,
      alignItems: 'center',
    },
    textInItem:{
      color:'#21254a'
    },
    button:{
        alignItems: 'center',
    }
  });