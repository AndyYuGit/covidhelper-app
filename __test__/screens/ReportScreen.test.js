import React from 'react';
import renderer from 'react-test-renderer';
import {ReportScreen} from "../../src/screens/ReportScreen";
jest.useFakeTimers()

jest.mock('@react-navigation/native', () => ({
    useIsFocused: jest.fn(),
  }));

test('renders correctly', () => {
    const tree = renderer.create(<ReportScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });