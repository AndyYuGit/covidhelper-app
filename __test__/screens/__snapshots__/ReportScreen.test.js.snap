// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#EEF0F4",
      "flexGrow": 1,
      "marginBottom": 0.1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "padding": 10,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
      />
    </View>
    <Text>
      Choose a report
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
      />
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#cba3ce",
        "height": 0.5,
        "width": "100%",
      }
    }
  />
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    ListHeaderComponent={[Function]}
    contentContainerStyle={
      Object {
        "flexGrow": 1,
      }
    }
    data={null}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <Text>
          Here are buildings about your selected locations with reported case at 
          :
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
